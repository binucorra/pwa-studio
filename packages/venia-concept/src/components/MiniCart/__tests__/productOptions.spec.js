import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import ProductOptions from '../productOptions';

test('it renders null when options are missing', () => {
    const tree = createTestInstance(<ProductOptions />);

    expect(tree).toMatchSnapshot();
});

test('it renders null when options is empty', () => {
    const props = {
        options: []
    };
    const tree = createTestInstance(<ProductOptions {...props} />);

    expect(tree).toMatchSnapshot();
});

test('it renders correctly when options is not empty', () => {
    const props = {
        options: [
            {
                label: 'Label 1',
                value: 'Value 1'
            },
            {
                label: 'Label 2',
                value: 'Value 2'
            }
        ]
    };
    const tree = createTestInstance(<ProductOptions {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});
