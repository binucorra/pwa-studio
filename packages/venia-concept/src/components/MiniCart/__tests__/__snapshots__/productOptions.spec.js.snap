// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly when options is not empty 1`] = `
<dl>
  <dt>
    Label 1
     : 
    Value 1
  </dt>
  <dt>
    Label 2
     : 
    Value 2
  </dt>
</dl>
`;

exports[`it renders null when options are missing 1`] = `null`;

exports[`it renders null when options is empty 1`] = `null`;
